### PR TITLE
BinaryOperatorSpacesFixer - Fix spaces around multiple exception catching (|)

### LINKED ISSUE #3207: BinaryOperatorSpacesFixer doesn't work with multiple exceptions in one catch block
PHP version: 7.1.9
PHP CS Fixer version: 2.8.0

In PHP 7.1 multiple exceptions can be specified in a single `catch` block using the `|` operator.

The BinaryOperatorSpacesFixer doesn't fix the spaces around `|` when used in a `catch` block.

Actual result:
```php
try {
    // some code
} catch (Exception1|       Exception2 $e) {
}
```

Expected result:
```php
try {
    // some code
} catch (Exception1 | Exception2 $e) {
}
```

### PR DESCRIPTION
**Pull request by ntzm**
closes #3207

### CODE REVIEW

File: tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php [hunk 1]
```diff
@@ -2072,6 +2072,23 @@ function A(){}[$a] = $a[$c];
...
+                    }
+                ',
+                '<?php
+                    try {
+                        // foo
+                    } catch (ExceptionOne|     ExceptionTwo $e) {
+                        // bar
+                    }
+                ',
+                ['operators' => ['|' => BinaryOperatorSpacesFixer::SINGLE_SPACE]],
```
Comment (GrahamCampbell) [inline tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php]: Should this operator not also be added to the defaults?

File: src/Tokenizer/TokensAnalyzer.php [hunk 2]
```diff
@@ -458,6 +458,7 @@ public function isBinaryOperator($index)
                 T_SR => true,                   // >>
                 T_SR_EQUAL => true,             // >>=
                 T_XOR_EQUAL => true,            // ^=
+                CT::T_TYPE_ALTERNATION => true, // |
```
[the remark this answers is not in the document]
Comment (ntzm) [reply src/Tokenizer/TokensAnalyzer.php]: Done

**Comment by SpacePossum** [comment]
On 2.2 the fix will look very different because of the rewrite of this fixer on 2.7. It would be nice to have this fix on 2.7, didn't check if that would be hard to do though.

**Comment by ntzm** [comment]
I can rebase to 2.7 if you want?

**Comment by ntzm** [comment]
I made a new PR for 2.2... Wasn't sure how to rebase properly

**Comment by SpacePossum** [comment]
thanks for the work @ntzm , I'll have a look on the 2.7 version as well :)

**Comment by ntzm** [comment]
Closing for #3215

**Pull request closed without merging**